client/src/app/components/auth/register/register.component.ts

```diff
@@ -28,7 +28,19 @@ export class RegisterComponent implements OnInit {
     });
   }
 
-  onSubmit() {
+  private arePasswordsEqual(): boolean {
+    if(this.registerForm.get('password')?.value === this.registerForm.get('confirmPassword')?.value) {
```

Comment:
Sth like this should be enough
`return this.registerForm.get('password')?.value === this.registerForm.get('confirmPassword')?.value`